### PR TITLE
add image-preferences blogpost

### PR DESCRIPTION
**Pull request by davidberenstein1957**
Congratulations! You've made it this far! Once merged, the article will appear at https://huggingface.co/blog. Official articles 
require additional reviews. Alternatively, you can write a community article following the process [here](https://huggingface.co/blog-explorers).

## Preparing the Article

You're not quite done yet, though. Please make sure to follow this process (as documented [here](https://github.com/huggingface/blog/tree/main?tab=readme-ov-file#how-to-write-an-article-)):

- [X] Add an entry to [_blog.yml](https://github.com/huggingface/blog/blob/main/_blog.yml).
- [x] Add a thumbnail. There are no requirements here, but there is a [template](https://github.com/huggingface/blog?tab=readme-ov-file#how-to-get-a-nice-responsive-thumbnail) if it's helpful.
- [X] Check you use a short title and blog path.
- [X] Upload any additional assets (such as images) to the Documentation Images [repo](https://huggingface.co/datasets/huggingface/documentation-images/tree/main/blog). This is to reduce bloat in the GitHub base repo when cloning and pulling. Try to have small images to avoid a slow or expensive user experience.
- [X] Add metadata (such as authors) to your `md` file. You can also specify `guest` or `org` for the authors.
- [X] Ensure the publication date is correct.
- [X] Preview the content. A quick way is to paste the markdown content in https://huggingface.co/new-blog. **Do not click publish**, this is just a way to do an early check.

Here is an example of a complete PR: https://github.com/huggingface/blog/pull/2382

## Getting a Review

Please make sure to get a review from someone on your team or a co-author.
Once this is done and once all the steps above are completed, you should be able to merge.
There is no need for additional reviews if you and your co-authors are happy and meet all of the above.

Feel free to add @pcuenca as a reviewer if you want a final check. Keep in mind he'll be biased toward light reviews
(e.g., check for proper metadata) rather than content reviews unless explicitly asked.

### CODE REVIEW

File: image-preferences.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+---
+title: "Hugging Face Community Releases an Open Preference Dataset for Text-to-Image Generation"
+thumbnail: /blog/assets/image_preferences/thumbnail.png
+authors:
+- user: davidberenstein1957
+- user: burtenshaw
+- user: dvilasuero
+- user: davanstrien
+- user: sayakpaul
+- user: linoyts
```
Comment (dvsrepo) [inline image-preferences.md:11]:
```suggestion
- user: linoyts
- user: Ameeeee
```

File: image-preferences.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+---
+# Hugging Face Community Releases an Open Preference Dataset for Text-to-Image Generation
+
+<details>
```
Comment (dvsrepo) [inline image-preferences.md:15]: this should go after the first paragraph

File: image-preferences.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+thumbnail: /blog/assets/image_preferences/thumbnail.png
+authors:
+- user: davidberenstein1957
+- user: burtenshaw
+- user: dvilasuero
+- user: davanstrien
+- user: sayakpaul
+- user: linoyts
+---
+# Hugging Face Community Releases an Open Preference Dataset for Text-to-Image Generation
```
Comment (dvsrepo) [inline image-preferences.md:13]:
```suggestion
# The Hugging Face Community Releases an Open Preference Dataset for Text-to-Image Generation
```

File: image-preferences.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+<details>
+  <summary>Unfamiliar with the Data is Better Together community?</summary>
+  <p>
+    Data is Better Together is a collaboration between 🤗 Hugging Face, 🏓 Argilla, and the Open-Source AI community. We aim to empower the open-source community to build impactful datasets collectively.
```
Comment (dvsrepo) [inline image-preferences.md:18]:
```suggestion
    [Data is Better Together](https://huggingface.co/data-is-better-together) is a collaboration between 🤗 Hugging Face and the Open-Source AI community. We aim to empower the open-source community to build impactful datasets collectively. You can follow the organization to stay up to date with the latest datasets, models, and community sprints.
```

File: image-preferences.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  </p>
+</details>
+
+The Data is Better Together community releases yet another important dataset for open source development. Due to the lack of open source datasets for text-to-image generation, we set out to release an Apache 2.0 licensed dataset for text-to-image generation. This dataset should contain 10K text-to-image preference pairs across common image generation categories, while using different model families and varying prompt complexities.
```
Comment (dvsrepo) [inline image-preferences.md:22]:
```suggestion
The Data is Better Together community releases yet another important dataset for open source development. Due to the lack of open preference datasets for text-to-image generation, we set out to release an Apache 2.0 licensed dataset for text-to-image generation. This dataset should contain 10K text-to-image preference pairs across common image generation categories, while using different model families and varying prompt complexities.
```

File: image-preferences.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+Impatient? All results can be found in [this collection on the Hugging Face Hub](https://huggingface.co/collections/data-is-better-together/image-preferences-675135cc9c31de7f912ce278) and code for pre- and post-processing can be found in [this GitHub repository](https://github.com/huggingface/data-is-better-together). Most importantly, there is a [ready-to-go preference dataset](https://huggingface.co/datasets/data-is-better-together/image-preferences-results-binarized) and a [flux-dev-lora-finetune](https://huggingface.co/data-is-better-together/image-preferences-flux-dev-lora). If you want to show your support already, don’t forget to like, subscribe and follow us before you continue reading further.
```
Comment (dvsrepo) [inline image-preferences.md:24]: The Impatient opening feels a bit disconnected, maybe something more explicit, like Want to jump into the dataset?

**Review by sdiazlor — COMMENTED** [review]
- What's the  final intention? Community or dataset release? Depending on that, I'd shorten the title with a more straightforward one
- I don't remember exactly where, but I recall some conversations mentioning that adding a short tldr section right at the beginning was useful for users to know what they can expect
- Mention to https://huggingface.co/data-is-better-together-contributor?

File: image-preferences.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+authors:
+- user: davidberenstein1957
+- user: burtenshaw
+- user: dvilasuero
+- user: davanstrien
+- user: sayakpaul
+- user: Ameeeee
+- user: linoyts
+---
+# Hugging Face Community Releases an Open Preference Dataset for Text-to-Image Generation
```
Comment (sayakpaul) [inline image-preferences.md:14]:
```suggestion
# Open Preference Dataset for Text-to-Image Generation by the 🤗 Community
```

File: image-preferences.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
+
+---
+title: "The Hugging Face Community Releases an Open Preference Dataset for Text-to-Image Generation"
+thumbnail: /blog/assets/image_preferences/thumbnail.png
```
Comment (sayakpaul) [inline image-preferences.md:4]: Probably missing currently?

File: image-preferences.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+Want to see the results? All results can be found in [this collection on the Hugging Face Hub](https://huggingface.co/collections/data-is-better-together/image-preferences-675135cc9c31de7f912ce278) and code for pre- and post-processing can be found in [this GitHub repository](https://github.com/huggingface/data-is-better-together). Most importantly, there is a [ready-to-go preference dataset](https://huggingface.co/datasets/data-is-better-together/image-preferences-results-binarized) and a [flux-dev-lora-finetune](https://huggingface.co/data-is-better-together/image-preferences-flux-dev-lora). If you want to show your support already, don’t forget to like, subscribe and follow us before you continue reading further.
+
+<details>
+  <summary>Unfamiliar with the Data is Better Together community?</summary>
+  <p>
+    [Data is Better Together](https://huggingface.co/data-is-better-together) is a collaboration between 🤗 Hugging Face and the Open-Source AI community. We aim to empower the open-source community to build impactful datasets collectively. You can follow the organization to stay up to date with the latest datasets, models, and community sprints.
+  </p>
+</details>
+
```
Comment (sayakpaul) [inline image-preferences.md:26]: We should have a mini section on the existing efforts in this area:

* Pick-a-Pic
* Fal
* https://huggingface.co/spaces/TIGER-Lab/GenAI-Arena
* https://artificialanalysis.ai/text-to-image/arena

File: image-preferences.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+## The input dataset
+
+To get a proper input dataset for this sprint, we started with some base prompts, which we cleaned, filtered for toxicity and injected with categories and complexities using synthetic data generation with [distilabel](https://github.com/argilla-io/distilabel). Lastly, we used Flux and Stable Diffusion models to generate the images. This resulted in the following dataset:
+
+<iframe
+src="https://huggingface.co/datasets/data-is-better-together/image-preferences/embed/viewer/default/train"
+  frameborder="0"
+  width="100%"
+  height="560px"
+></iframe>
```
Comment (sayakpaul) [inline image-preferences.md:36]: If we're not already, can we comment a bit more about the different subjects, complexity levels of the prompts involved?

File: image-preferences.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+[Imgsys](https://imgsys.org/) is a generative image model arena hosted by [fal.ai](http://fal.ai), where people provide prompts and get to choose between two model generations to provide a preference. Sadly, the generated images are not published publicly, however, [the associated prompts are hosted on Hugging Face](https://huggingface.co/datasets/fal/imgsys-results). These prompts represent real-life usage of image generation containing good examples focused on day-to-day generation, but this real-life usage also meant it contained duplicate and toxic prompts, hence we had to look at the data and do some filtering.
+
+### Reducing toxicity
+
+We aimed to remove all NSFW prompts and images from the dataset before starting the community. We settled on a multi-model approach where we used two text-based and two image-based classifiers as filters. Post-filtering, we decided to do a manual check of each one of the images to make sure no toxic content was left, luckily we found our approach had worked.
+We used the following pipeline:
+- Classify images as NSFW
+- Remove all positive samples
+- Argilla team manually reviews the dataset
+- Repeat based on review
```
Comment (sayakpaul) [inline image-preferences.md:49]:
```suggestion
We aimed to remove all NSFW prompts and images from the dataset before starting the community. We settled on a multi-model approach where we used two text-based and two image-based classifiers as filters. Post-filtering, we decided to do a manual check of each one of the images to make sure no toxic content was left, luckily we found our approach had worked.

We used the following pipeline:
- Classify images as NSFW
- Remove all positive samples
- Argilla team manually reviews the dataset
- Repeat based on review
```

File: image-preferences.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+### Synthetic prompt enhancement
+
+Data diversity is important for data quality, which is why we decided to enhance our dataset by synthetically rewriting prompts based on various categories and complexities. This was done using a [distilabel pipeline](https://github.com/huggingface/data-is-better-together/blob/main/community-efforts/image_preferences/01_synthetic_data_generation_total.py).
+
+#### Prompt categories
+
+[InstructGPT](https://arxiv.org/pdf/2203.02155) describes foundational task categories for text-to-text generation but there is no clear equivalent of this for text-to-image generation. To alleviate this, we used two main sources as input for our categories: [google/sdxl](https://huggingface.co/spaces/google/sdxl/blob/main/app.py) and [Microsoft](https://www.microsoft.com/en-us/bing/do-more-with-ai/ai-art-prompting-guide/ai-genres-and-styles?form=MA13KP). This led to the following main categories:  ["Cinematic", "Photographic", "Anime", "Manga", "Digital art", "Pixel art", "Fantasy art", "Neonpunk", "3D Model", “Painting”, “Animation” “Illustration”]. On top of that we also chose some mutually exclusive, sub-categories to allow us to further diversify the prompts.
```
Comment (sayakpaul) [inline image-preferences.md:57]: Could include a table / pie-chart to visually convey the distribution of the categories.

Comment (davidberenstein1957) [reply image-preferences.md:67]: @sayakpaul they are more or less equally distributed based on random sampling so I will mention that.

File: image-preferences.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+- Tie: Photographic, Animation
+- FLUX-dev better: 3D Model, Anime, Manga
+- SD3.5-XL better: Cinematic, Digital art, Fantasy art, Illustration, Neonpunk, Painting, Pixel art
+
+### Model-finetune
+
+To verify the quality of the dataset, while not spending too much time and resources we decided to do a [Dreambooth](https://dreambooth.github.io/) fine-tune of the [black-forest-labs/FLUX.1-dev](https://huggingface.co/black-forest-labs/FLUX.1-dev) model. We did a supervised fine-tuning, where we only included the chosen sample as expected completions. Interestingly, the chosen fine-tuned models perform much better in art and cinematic scenarios where it was initially lacking! You can [test the fine-tuned adapter here](https://huggingface.co/data-is-better-together/image-preferences-flux-dev-lora).
```
Comment (sayakpaul) [inline image-preferences.md:104]: Maybe spend one sentence to note that it's a LoRA fine-tune and not a DreamBooth fine-tune as we're NOT following any DreamBooth-specific things here (keywords, prior-preservation, etc.). 

So, best to say LoRA fine-tune. 

Additionally, I would mention the script that was used to do this as I think the definitely deserves a mention here.

File: image-preferences.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+
+![model-finetune](/blog/assets/image_preferences/model_finetune.png)
```
Comment (sayakpaul) [inline image-preferences.md:106]: We're saying that the SFT fine-tune leads to better quality results but we're not really showing it here in the figure. Perhaps 2/3 examples would actually give the readers a bit of a boost to consider this even more.

File: image-preferences.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+
+![leaderboard](/blog/assets/image_preferences/leaderboard.png)
+
+## What is next?
+
+After another successful community sprint, we will continue organising them on the Hugging Face Hub. Make sure to follow [the Data Is Better Together organisation](https://huggingface.co/data-is-better-together) to stay updated. We also encourage community members to take action themselves and are happy to guide and reshare on socials and within the organisation on the Hub. Anyone can contribute in several ways:
+- Join and participate in other sprints.
+- Propose your own sprints or requests for high quality datasets.
+- Fine-tune models on top of [the preference dataset](https://huggingface.co/datasets/data-is-better-together/image-preferences-results-binarized). One idea would be to do a full SFT fine-tune of SDXL or FLUX-schnell. Another idea would be to do a DPO/ORPO fine-tune.
+- Evaluate the improved performance of [the LoRA adapter](https://huggingface.co/data-is-better-together/image-preferences-flux-dev-lora) compared to the original SD3.5-XL and FLUX-dev models.
```
Comment (sayakpaul) [inline image-preferences.md]:
```suggestion
After another successful community sprint, we will continue organising them on the Hugging Face Hub. Make sure to follow [the Data Is Better Together organisation](https://huggingface.co/data-is-better-together) to stay updated. We also encourage community members to take action themselves and are happy to guide and reshare on socials and within the organisation on the Hub. Anyone can contribute in several ways:
- Join and participate in other sprints.
- Propose your own sprints or requests for high quality datasets.
- Fine-tune models on top of [the preference dataset](https://huggingface.co/datasets/data-is-better-together/image-preferences-results-binarized). One idea would be to do a SFT fine-tune of smaller models like [SDXL](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0) or [PixArt-Sigma](https://huggingface.co/PixArt-alpha/PixArt-Sigma-XL-2-1024-MS), verifying if its qualitative aspects are improved. 
- Fine-tune with preference alignment techniques as opposed to SFT. One could consider techniques like [MaPO](https://arxiv.org/abs/2406.06424) or [Diffusion DPO](https://arxiv.org/abs/2311.12908).
- Evaluate the improved performance of [the LoRA adapter](https://huggingface.co/data-is-better-together/image-preferences-flux-dev-lora) compared to the original SD3.5-XL and FLUX-dev models.
```

**Review by sayakpaul — COMMENTED** [review]
Thanks for working on this! Left a couple of comments.

**Review by dvsrepo — APPROVED** [review]

**Review by Vaibhavs10 — APPROVED** [review]

**Pull request merged**